opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchDataSourceFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.storage;
+
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.datasource.model.DataSource;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.opensearch.client.OpenSearchClient;
+import org.opensearch.sql.storage.DataSourceFactory;
+
+@RequiredArgsConstructor
+public class OpenSearchDataSourceFactory implements DataSourceFactory {
+
+  /** OpenSearch client connection. */
+  private final OpenSearchClient client;
+
+  private final Settings settings;
```

Comment:
In future, can we refactor this to a configMetadata class instead of passing down Settings object. Is this because, if we change any transient settings will it always give the latest information?

Reply:
yes, we can add our abstraction. my understanding is settings will be updated.